### DISCUSSION TITLE
Add and populate and ID column

### DISCUSSION THREAD

**Discussion by testerez** [prisma/prisma#19339] [Q&A] [answered]
Hello,
I'm using prisma with PostgeSQL  and I want to add an "id" column on a table that previously had an id composed of two columns with `@@id([a, b])`.

I'm adding this new "id" column like this:
```
id String @default(cuid())
```

But since this table already has values, the migration fails because it can't automatically populate missing ids. What would be the easiest way to fix this?

**Comment by ludralph** [comment]
Hi @testerez 👋 

Can you please clarify if you want the new `id` column to be the only id field or if you want to add it to the existing ids `a`and `b`.

Also, you can follow these steps to add the id field:

1. add the id field and make it optional then create a migration for it.
2. Write an sql script to add records to the missing  ids
3. create another migration that removes the optional `id` field and makes it mandatory

**Comment by testerez** [comment, marked as answer]
Hey @ludralph, thanks for your answer. In my case, I wanted `id` to be the only id field.

The solution you are suggesting looks a lot like what I ended up doing. This is a sniped of the migration I wrote:

```sql
-- AlterTable
ALTER TABLE "Quote" DROP CONSTRAINT "Quote_pkey",
  ADD COLUMN "id" TEXT;
-- populate missing ids
UPDATE "Quote"
SET "id" = CONCAT(
    'c',
    CAST(
      FLOOR(
        EXTRACT(
          EPOCH
          FROM NOW()
        ) * 1000
      ) AS VARCHAR
    ),
    CAST(FLOOR(RANDOM() * 1000000000) AS VARCHAR)
  )
WHERE id IS NULL;
-- Add primary key
ALTER TABLE "Quote"
ALTER COLUMN "id"
SET NOT NULL,
  ADD CONSTRAINT "Quote_pkey" PRIMARY KEY ("id");
```

**Reply by ludralph** [reply]
Thanks for confirming what worked for you.
